### PR TITLE
[Fix](Nereids)fix aggregation result error when child is empty relation

### PR DESCRIPTION
**Pull request by sohardforaname**
# Proposed changes

Issue Number: close #xxx

## Problem summary

When a sql as select max(v) from t where 1=2 running, after analysis, v is nullable so max(v) is nullable, it will return 0 instead of null, but the latter is expected.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by morrySnow — COMMENTED** [review]
please add regression test~

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.8 seconds
 load time: 470 seconds
 storage size: 17123356285 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221209092922_clickbench_pr_60961.html

**Review by 924060929 — DISMISSED** [review]

**Comment by morrySnow** [comment]
from regression test, we need to do
1. add back the original constructor
2. refactor agg function registry to block read the constructor with alwaysNullable params

**Comment by morrySnow** [comment]
please rebase to newest master code to pass FE UT

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Min.java [hunk 1]
```diff
@@ -32,15 +31,19 @@
...
+        this(false, false, child);
     }
 
     public Min(boolean isDistinct, Expression arg) {
-        super("min", false, arg);
+        this(isDistinct, false, arg);
+    }
+
+    public Min(boolean isDistinct, boolean isAlwaysNullable, Expression arg) {
+        super("min", isAlwaysNullable, isDistinct, arg);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Min.java:46]: maybe all alwaysnullable constructor should be private

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/FillUpMissingSlots.java [hunk 2]
```diff
@@ -171,6 +172,9 @@ public void resolve(Expression expression) {
                         throw new AnalysisException("Aggregate functions in having clause can't be nested: "
                                 + expression.toSql() + ".");
                     }
+                    if (groupByExpressions.isEmpty() && expression instanceof NullableAggregateFunction) {
+                        expression = ((NullableAggregateFunction) expression).withAlwaysNullable(true);
+                    }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/FillUpMissingSlots.java:177]: chould we make a new rule to process nullable only?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Avg.java [hunk 3]
```diff
@@ -64,13 +67,18 @@ protected List<DataType> intermediateTypes() {
     @Override
     public AggregateFunction withDistinctAndChildren(boolean isDistinct, List<Expression> children) {
         Preconditions.checkArgument(children.size() == 1);
-        return new Avg(isDistinct, children.get(0));
+        return new Avg(isDistinct, isAlwaysNullable, children.get(0));
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Avg.java:70]: 什么情况下 isAlwaysNullable=false？

**Comment by morrySnow** [comment]
duplicated with #15547

**Pull request closed without merging**